plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -889,35 +589,38 @@ open class DefaultPageCreator(
     }
 
     protected open fun TagWrapper.toHeaderString() = this.javaClass.toGenericString().split('.').last()
+}
 
-    private val List<Documentable>.sourceSets: Set<DokkaSourceSet>
-        get() = flatMap { it.sourceSets }.toSet()
+internal val List<Documentable>.sourceSets: Set<DokkaSourceSet>
+    get() = flatMap { it.sourceSets }.toSet()
 
-    private val List<Documentable>.dri: Set<DRI>
-        get() = map { it.dri }.toSet()
+internal val List<Documentable>.dri: Set<DRI>
+    get() = map { it.dri }.toSet()
 
-    private val Documentable.groupedTags: GroupedTags
-        get() = documentation.flatMap { (pd, doc) ->
-            doc.children.map { pd to it }.toList()
-        }.groupBy { it.second::class }
+internal val Documentable.groupedTags: GroupedTags
+    get() = documentation.flatMap { (pd, doc) ->
+        doc.children.map { pd to it }.toList()
+    }.groupBy { it.second::class }
 
-    private val List<Documentable>.groupedTags: GroupedTags
-        get() = this.flatMap {
-            it.documentation.flatMap { (pd, doc) ->
-                doc.children.map { pd to it }.toList()
-            }
-        }.groupBy { it.second::class }
+internal val Documentable.descriptions: SourceSetDependent<Description>
+    get() = groupedTags.withTypeUnnamed()
 
-    private val Documentable.descriptions: SourceSetDependent<Description>
-        get() = groupedTags.withTypeUnnamed()
+internal val Documentable.customTags: Map<String, SourceSetDependent<CustomTagWrapper>>?
```

Comment:
It makes the empty check a little less laconic
```kt
val customTags = documentable.customTags
if (customTags.isEmpty()) return
```
instead of
```kt
val customTags = documentable.customTags ?: return
```
But if it is a good practice…

Reply:
It is, I promise I didn't make it up :sweat_smile: I believe it comes from the "Effective Java" book initially

Nullability adds yet another state to the collection - it can be null and can be empty, so you have to check for both. It's less severe in Kotlin with null-safety and `map?.isEmpty()` checks, but in Java it would've been a bigger problem and would lead to an NPE sooner or later. 

It doesn't matter much in this particular case, but smart casts are not always possible (if it's a val within an object) and constant null checking can add more boilerplate code than if it simply returned an empty collection

We can discuss this during sync if you want

Reply:
Btw, if you wanted, you could do 
```
map.takeIf { it.isNotEmpty() } ?: return
```
but I would personally keep it simple :)